packages/connect/src/api/cancelCoinjoinAuthorization.ts

```diff
@@ -0,0 +1,25 @@
+import { AbstractMethod } from '../core/AbstractMethod';
+import { getFirmwareRange } from './common/paramsValidator';
+import { PROTO } from '../constants';
+
+export default class CancelCoinjoinAuthorization extends AbstractMethod<
+    'cancelCoinjoinAuthorization',
+    PROTO.CancelAuthorization
+> {
+    init() {
+        this.firmwareRange = getFirmwareRange(this.name, null, this.firmwareRange);
+        this.info = 'Cancel Coinjoin Authorization';
+    }
+
+    async run() {
+        const cmd = this.device.getCommands();
+
+        if (!(await cmd.preauthorize(false))) {
```

Comment:
use `this.preauthorized` field to use different variant of device state validation (not by address), see AbstractMethod.preauthorized explanation and AuthorizeCoinjoin method

Reply:
yes, `preauthorized` is a CommonParam it should be set in the `init` function.

for this particular case it could fallback to `true` if it's not set:

```
this.preauthorized = typeof  payload.preauthorized === 'boolean' ?  payload.preauthorized : true;
```


it would be also nic to extend `authorizeCoinjoin` e2e test to check if it really do what is expected.

scenario:
- authorizeCoinjoin
- getOwnershipProof (no button request expected)
- unauthorizeCoinjoin
- getOwnershipProof (button request expected)

Reply:
Hmm, there is no button request currently. But I will add a test for the method just to see if it works

Reply:
6ea6346ee4d4a8617d2379363b5a2f621fdb0bb2

Reply:
hm, now there is too much validation :) you don't have to use `validateParams` since you are doing the same check below, no?
 `typeof payload.preauthorized === 'boolean'`

Reply:
Right, it got changed to optional ea28763c8